SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -413,10 +467,33 @@ void USpatialSender::SendRPC(TSharedRef<FPendingRPCParams> Params)
 		// Target object was destroyed before the RPC could be (re)sent
 		return;
 	}
-
 	UObject* TargetObject = Params->TargetObject.Get();
+
+	if (AActor* TargetActor = Cast<AActor>(TargetObject))
+	{
+		USpatialNetConnection* SpatialConnection = NetDriver->GetSpatialOSNetConnection();
+		if (!NetDriver->GetActorChannelByEntityId(PackageMap->GetEntityIdFromObject(TargetActor)))
+		{
+			NetDriver->CreateActorChannel(TargetActor, SpatialConnection);
+		}
+
+		if (USpatialActorChannel* Channel = NetDriver->GetActorChannelByEntityId(PackageMap->GetEntityIdFromObject(TargetActor)))
+		{
+			if (Channel->bCreatingNewEntity)
+			{
+				OutgoingOnCreateEntityRPCs.FindOrAdd(TargetActor).Add(Params);
+				return;
+			}
+		}
+		else
+		{
+			UE_LOG(LogSpatialSender, Warning, TEXT("Failed to create an Actor Channel for %s."), *TargetActor->GetName());
+		}
+	}
+
 	if (PackageMap->GetUnrealObjectRefFromObject(TargetObject) == FUnrealObjectRef::UNRESOLVED_OBJECT_REF)
 	{
+		// This is where we'll serialize this RPC and queue it to be added on entity creation
 		UE_LOG(LogSpatialSender, Verbose, TEXT("Trying to send RPC %s on unresolved Actor %s."), *Params->Function->GetName(), *TargetObject->GetName());
 		QueueOutgoingRPC(TargetObject, Params);
```

Comment:
Hmm, could we be queuing twice? This logic maybe seems wrong now if were queuing the ones above.

Reply:
There's a return statement there, so we can't be queuing twice.